hydroflow_lang/src/graph/ops/join.rs

```diff
@@ -16,6 +17,45 @@ use syn::parse_quote;
 /// recv_iter(vec![("hello", "cleveland")]) -> [1]my_join;
 /// my_join -> for_each(|(k, (v1, v2))| println!("({}, ({}, {}))", k, v1, v2));
 /// ```
+///
+/// `join` can also be provided with a single generic lifetime argument, either `'epoch` or
```

Comment:
Actually for join we'd like to control the lifetime on the two tables separately (table-table join, stream-table join, stream-stream join).